ui/panels-plugin/src/plugins/trace-table/TraceTablePanel.tsx

```diff
@@ -0,0 +1,59 @@
+// Copyright 2024 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { PanelProps, useDataQueries } from '@perses-dev/plugin-system';
+import { Box, Skeleton } from '@mui/material';
+import { ErrorAlert, useChartsTheme } from '@perses-dev/components';
+import { DataTable } from './DataTable';
+import { TraceTableOptions } from './trace-table-model';
+
+export type TraceTableProps = PanelProps<TraceTableOptions>;
+
+export function TraceTablePanel(props: TraceTableProps) {
+  const { contentDimensions } = props;
+  const chartsTheme = useChartsTheme();
+  const { isFetching, isLoading, queryResults } = useDataQueries('TraceQuery');
+  const contentPadding = chartsTheme.container.padding.default;
+
+  if (contentDimensions === undefined) {
+    return null;
+  }
+
+  if (isLoading || isFetching) {
+    return (
+      <Box
+        sx={{ display: 'flex', alignItems: 'center', justifyContent: 'center' }}
+        width={contentDimensions.width}
+        height={contentDimensions.height}
+      >
+        <Skeleton
+          variant="text"
+          width={contentDimensions.width - 20}
+          height={contentDimensions.height / 2}
+          aria-label="Loading..."
+        />
+      </Box>
+    );
+  }
+
+  const tracesFound = queryResults.some((traceData) => (traceData.data?.traces ?? []).length > 0);
+  if (!tracesFound) {
+    return <ErrorAlert error={{ name: 'No traces found.', message: 'No traces found.' }} />;
```

Comment:
I believe no traces found should not be an error alert but ok for the DataTable. A "No traces found" message centered in regular typography would be preferred. 
Like we have in the dashboards table 
![image](https://github.com/perses/perses/assets/10258608/bcf5efcf-3e69-4431-80e4-bcb0c420d00a)

Reply:
IMO that should be the same message as when no timeseries found for graphs, for consistency. Ideally this would even be something a plugin developer should not have to deal with. 


In the specific case of the traces explorer now.
In case there's no data, I would see a message no data in the scatter plot and no table panel at all WDYT ? 
Same for errors and loading.
Since the beginning I feel like what you're doing is kind of a legend and that's what a legend would do

Reply:
> IMO that should be the same message as when no timeseries found for graphs, for consistency. Ideally this would even be something a plugin developer should not have to deal with.

I like the idea, I've created https://github.com/perses/perses/issues/2077 to track this.

> In the specific case of the traces explorer now. In case there's no data, I would see a message no data in the scatter plot and no table panel at all WDYT ? Same for errors and loading. 

If #2077 gets implemented, maybe the panel in the explorer page could be a "meta-panel", containing the ScatterPlot and the TraceTable, and handling the loading/error/no data state at the parent Panel.